plugins/mongo_db_plugin/mongo_db_plugin.cpp

```diff
@@ -1427,39 +1435,44 @@ void mongo_db_plugin_impl::init() {
          }
 
          try {
+            // Due to the vast amounts of data, we suggest:
+            //   1. enableSharding database (default to EOS)
+            //   2. shardCollection: blocks, action_traces, transaction_traces, especially action_traces
+            //   3. Use compound index with shard key (default to _id), to improve query performance.
+
             // blocks indexes
             auto blocks = mongo_conn[db_name][blocks_col];
-            blocks.create_index( bsoncxx::from_json( R"xxx({ "block_num" : 1 })xxx" ));
-            blocks.create_index( bsoncxx::from_json( R"xxx({ "block_id" : 1 })xxx" ));
+            blocks.create_index( bsoncxx::from_json( R"xxx({ "block_num" : 1, "_id" : 1 })xxx" ));
+            blocks.create_index( bsoncxx::from_json( R"xxx({ "block_id" : 1, "_id" : 1 })xxx" ));
```

Comment:
Do you have any feel for how much larger the indexes are when _id is added?

Reply:
```
use EOS
db.action_traces.stats()
```
shows:
```
"indexSizes" : {
				"_id_" : 18581565440,
				"trx_id_1" : 16938258432,
				"block_num_1" : 4512980992,
				"block_num_1__id_1" : 21147549696
```

`block_num_1__id_1` is 16,634,568,704 bytes larger than `block_num_1`, close to "_id_" : 18,581,565,440.

Reply:
I wonder if the plugin should even create any indexes and let that be left to the user. However, I'm inclined to accept this PR if you rebase to `develop`, remove the try-catch changes (or provide some article/test/proof it makes a difference), and update the comment.

Thanks.